Refactor of main and beacon activites, and others

Sonderful opened the pull request:
This is a fairly large change, but the intention is not to change any behavior for users, rather it aims to refactor the code slightly, from what appears to be mostly auto-converted code.

* Replaced findViewById with synthetics in MainActivity and BeaconActivity where appropriate.
* Made several changes to those classes, and other related classes, to make things a bit more null-safe, a bit tidier, and a bit more idiomatic.
* Reduced visiblity of the Ranging button click handler by registering the click listener in code instead of the layout file.
* Increased visiblity of the TTSManager onInitListener because it was referenced from the unit tests, but not compiling for me.

I'm happy to try to answer questions if there are any. I'm on Mattermost if a longer discussion or explanation is wanted.

Commit:
Refactor of main and beacon activites, and others

* Replaced findViewById with synthetics in MainActivity and BeaconActivity where appropriate.
* Made several changes to those classes, and other related classes, to make things a bit more null-safe, a bit tidier, and a bit more idiomatic.
* Reduced visiblity of the Ranging button click handler by registering the click listener in code instead of the layout file.
* Increased visiblity of the TTSManager onInitListener because it was referenced from the unit tests, but not compiling for me.

skraynick commented:
@Sonderful please check the mattermost chat. I believe we were axing the beacon stuff. I will hold off until you respond.... thanks.

Commit:
Removing unused beacon feature

Removed the activities, layouts, strings, preferences, utils, dependencies, and documentation references to the beacon feature.
This commit can be reverted in order to revive the feature in the future, if desired.

skraynick commented on mobile/src/main/java/mycroft/ai/SettingsActivity.kt:
```diff
@@ -247,20 +214,7 @@ class SettingsActivity : AppCompatPreferenceActivity() {
                     }
                 }
             } else if (preference is SwitchPreference) {
```
Do we need this?

Sonderful replied:
There are currently no switch preferences, so this can be removed, along with the concept of a switch preference.

It felt like the kind of thing that would almost certainly be needed in the future, so I didn't remove it.

I'm working on another change (dependency upgrades/cleanup) but I could add in another commit to remove the unused preference type. It'll be easy enough to rewrite or restore if it's ever needed.

skraynick commented on mobile/src/main/res/layout/content_main.xml:
```diff
@@ -34,13 +34,13 @@
         android:layout_alignParentStart="true"
-        android:layout_marginTop="40dp"
-        android:layout_marginBottom="60dp"/>
+        android:layout_marginTop="@dimen/card_list_margin_top"
```
Nice! Love to see this... :)

skraynick commented:
@Sonderful please check the 1 comment... thanks.

Commit:
Removing no-op switch preference handling

Handling was only ever used to handle Beacon preferences as special cases. Other switch preferences are handled in the default manner.
